### PR TITLE
Adds impl FromSkyhashBytes for RawString and a TODO

### PR DESCRIPTION
**Pull request by emchristiansen**
This is just a code sketch, as I'm a bit unclear on a couple things:

1. I'm not sure why this wasn't already included (and I assume you have a good reason).
1. I don't understand why you're encoding the key / value data the way you are. For example, in line 475 you seem to assume primitives are encoded as UTF8 strings, which seems very odd. Rather I'd expect everything in the DB to be (arbitrary, not UTF8) `Vec<u8>`s. I think you can convert all these primitives to bytes using functions like `_.to_le_bytes()`.

Relatedly, I'm not sure why you need the `RawString` type instead of just implementing the appropriate traits for `Vec<u8>`.

### CODE REVIEW

**Comment by glydr** [comment]
[CLA assistant check](https://cla.skytable.io/skytable/client-rust?pullRequest=16) <br/>All committers have signed the CLA.

File: src/types.rs [hunk 1]
```diff
@@ -459,6 +469,9 @@ macro_rules! impl_from_skyhash {
         $(impl FromSkyhashBytes for $ty {
             fn from_element(element: Element) -> SkyResult<$ty> {
                 let ret = match element {
+                    // Note: We're assuming primitive types are internally
+                    // represented as strings, which we then parse.
+                    // TODO: Represent primitives more efficiently.
```
Comment (ohsayan) [inline src/types.rs:474]: This is just for convenience. We can simply do a `to_string` but it's not applicable for other types in this specific shorthand. Second, no, primitive types are **not** represented as strings internally. The formatting infrastructure is used to produce strings for types like integers.

**Review by ohsayan — CHANGES_REQUESTED** [review]

File: src/types.rs [hunk 2]
```diff
@@ -399,6 +399,16 @@ impl<'a> IntoSkyhashBytes for &'a RawString {
...
+impl FromSkyhashBytes for RawString {
+    fn from_element(element: Element) -> SkyResult<RawString> {
+        let e = match element {
+            Element::Binstr(bstr) => bstr.into(),
+            _ => return Err(Error::ParseError(BAD_ELEMENT.to_owned())),
+        };
+        Ok(e)
+    }
+}
+
```
Comment (ohsayan) [inline src/types.rs:411]: `RawString` is simply a wrapper around `Vec<u8>` and only makes sense for sending queries that contain binary data and not reading them. Is there any specific reason you want to parse into a `RawString` and not a `Vec<u8>`?

**Review by ohsayan — CHANGES_REQUESTED** [review]

**Comment by ohsayan** [comment]
> I don't understand why you're encoding the key / value data the way you are. For example, in line 475 you seem to assume primitives are encoded as UTF8 strings,

The `FromSkyhashBytes` trait is more of an application layer parsing impl than anything related to how Skytable itself stores data. It is provided for convenience. Since currently the database allows storing a string/binary as a key/value and permutations of the same, you might have done something like this in your application code:

```rust
db.set("myint", "100").unwrap();
```

Now let's say your table stores: `string` key -> `binary` value. If you did a parse into the actual protocol level type, you'd run this:

```rust
let protocol_element: Element = db.get("myint").unwrap();
```

This would give you `Element::Binary`, but you actually wanted an integer. This is why the `FromSkyhashBytes` trait exists to make it convenient for you to simply run this:

```rust
let myint: u8 = db.get("myint").unwrap();
```

I hope this clears things up a bit :)

**Pull request closed without merging**